From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 21 Jul 2016 11:36:50 +0200
Subject: Don't use g_type_init on recent glib
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

it's not needed and avoids

test-cups-pk.c: In function ‘main’:
test-cups-pk.c:80:9: error: ‘g_type_init’ is deprecated
[-Werror=deprecated-declarations]
         g_type_init ();
---
 src/main.c         | 2 ++
 src/test-cups-pk.c | 2 ++
 src/test-cups.c    | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/src/main.c b/src/main.c
index ced7525..4f7f6e8 100644
--- a/src/main.c
+++ b/src/main.c
@@ -145,7 +145,9 @@ main (int argc, char **argv)
         cph_main data;
         guint    owner_id;
 
+#if ! GLIB_CHECK_VERSION(2, 36, 0)
         g_type_init ();
+#endif
 
         memset (&data, 0, sizeof (data));
 
diff --git a/src/test-cups-pk.c b/src/test-cups-pk.c
index 58f002d..83b0380 100644
--- a/src/test-cups-pk.c
+++ b/src/test-cups-pk.c
@@ -77,7 +77,9 @@ main (int argc, char **argv)
         gboolean           ret;
         GError            *error;
 
+#if ! GLIB_CHECK_VERSION(2, 36, 0)
         g_type_init ();
+#endif
 
         error = NULL;
         proxy = cph_iface_mechanism_proxy_new_for_bus_sync (
diff --git a/src/test-cups.c b/src/test-cups.c
index cf68467..a7500e9 100644
--- a/src/test-cups.c
+++ b/src/test-cups.c
@@ -31,7 +31,9 @@ main (int argc, char **argv)
 {
         CphCups *cups;
 
+#if ! GLIB_CHECK_VERSION(2, 36, 0)
         g_type_init ();
+#endif
 
         cups = cph_cups_new ();
 
